src/actions/assets/F3D.js

```diff
@@ -0,0 +1,47 @@
+import AssetActionPlugin from "../AssetActionPlugin";
+import URI from 'urijs';
+import i18n from "../../i18n";
+
+// obj & ply files are usually with mime-type text/plain 
+const F3D_SUPPORTED_TYPES = [
+  'model/gltf-binary', 
+  'model/gltf+json',
+  'application/fbx',
+];
+// below is usually text/plain
+const F3D_SUPPORTED_FILEEXTS = ['.obj', '.ply', '.fbx', '.glb', '.gltf'];
+export default class F3D extends AssetActionPlugin {
+
+  get show() {
+    return this.component.isBrowserProtocol && (
+      F3D_SUPPORTED_TYPES.includes(this.asset.type)
+      || F3D_SUPPORTED_FILEEXTS.map(
+        f => this.asset.href.toLowerCase().includes(f)
+      ).some(e => e)
```

Comment:
This would also match for https://www.objects.com/index.html
I think this should use urijs to get the path component and then do an endsWith check.

Reply:
I did not know about [objects.com](https://www.objects.com/) - site is under maintenance at the moment, as well as a few previous versions according to wayback machine so I cannot explore it. Is this an open-source 3d-formats-viewer implementation like f3d?

Reply:
That were just examples for potential false positives because the matching is potentially done on the full URL.